### ISSUE TITLE
Spotify Plugin Error

### ISSUE THREAD

**Issue by mckennajones** [tjmarkham/win10widgets#99]
Unfortunately, the Spotify plugin is not working on my machine. I have spotify running but the plugin is showing: "Error: Spotify Plugin not working". I have attached a screenshot of that, and the errors I am seeing in rainmeter. This is on version 4.1.0 of rainmeter.

Let me know what other info would be useful to you.

![win10widgetspotifyerror](https://user-images.githubusercontent.com/7456446/36552850-bdec6b34-17af-11e8-807b-d2cd7a96aa81.PNG)
![rainmetererrors](https://user-images.githubusercontent.com/7456446/36552851-be0c2bc2-17af-11e8-846f-465ca0fd2abe.PNG)

**Comment by BrendanTCC** [comment]
You're not the only who who gets this bug. I managed to find a fix a couple of days ago, here's the link:
https://forum.rainmeter.net/download/file.php?id=14061&sid=eafd4757ea966bbcc931bdb48b404342
The original plugin author also includes their own version of the spotify player with this update, just like with every other version, just for the record.

**Comment by mckennajones** [comment]
@BrendanTCC Right on, thanks so much. Works like a charm. Kudos to you for this sweet project btw!

**Comment by BrendanTCC** [comment]
I have nothing to do with Win 10 Widgets, nor the Spotify Plugin. I just so happened to have the same issue as you when I first got the Win 10 Widgets skin on my laptop (which was a few weeks ago), and I searched for a fix only a few days ago.

**Comment by mckennajones** [comment]
Whoops my mistake. Well thanks anyways.

**Comment by BrendanTCC** [comment]
No worries! =)

**Comment by mckennajones** [comment]
Looks like my Spotify updated (ver 1.0.75.483.g7ff4a0dc), and it's not working anymore unfortunately :(

**Comment by BrendanTCC** [comment]
I'll have a look for a fix for you. =)

**Comment by JoshWay** [comment]
Man... it's the little things that count sometimes in life.  Thank you for this!!

**Comment by BrendanTCC** [comment]
No worries! Heads up though, turns out Spotify changed the window class name, so that's why the plugin no longer even thinks the app exists. .raptor may release an update soon, though, so keep an eye on this forum:
https://forum.rainmeter.net/viewtopic.php?t=17077
=)

**Comment by tjhrulz** [comment]
That class name issue has nothing to do with .raptors Spotify plugin, it has to do with the NowPlaying measure built into Rainmeter (Which is out of date with the version of Rainmeter bundled with this). Also it is not something easily fixed as Spotify did not change the class name, they forgot to add one all together. This means it ended up with a default class name that is the same as probably 100+ other apps on your computer. Until Spotify fixes this there is no good and easy way to fix it. The workaround currently is to use .raptors plugin for player commands as well, but that will only work for premium users.

**Comment by Meowskers5** [comment]
Not the same exact issue, but I just downloaded the Win10Widgets today and the Spotify widget wouldn't work (no data). I edited the widget text and changed the following:

;PlayerName=Spotify
PlayerName=SpotifyPlugin.dll

This made the plugin work.  Thought I'd add a comment here in case it helps someone else out.

**Comment by chrispyyycrunch** [comment]
^^^ What widget text did you change, could you explain further, i also just installed it.

**Comment by Meowskers5** [comment]
For me it was Spotify-Small.ini, but I'm guessing the same edit will work for all three Spotify widgets.  Just select the widget and click the 'Edit' button.  I changed the 'Playername' directive to be 'SpotifyPlugin.dll' instead of 'Spotify'.

Edit: It is in Win10Widgets/Spotify/Spotify-Small.ini.  The other Spotify ini files reference this one.

**Comment by tjhrulz** [comment]
@Meowskers5 That is not a valid syntax for NowPlaying so there is no way that actually worked, the fix you need to make is a little different and I have previously outlined it in the forums: https://forum.rainmeter.net/viewtopic.php?p=145730#p145730

**Comment by Meowskers5** [comment]
@tjhrulz What I detailed is DIRECTLY a fix for the issue. Meaning if anyone had performed what I stated, their Win10Widget Spotify Plugin would start working again.  'Playername="SpotifyPlugin.dll" ' is obviously a known directive because even .raptor uses it in his SpotifyPlayer Version=1.1.0.0.  That's where I got the idea to try it from.  It might be deprecated or something, but to act as if I am just making things up is a little absurd.  After looking at this further, I have figured out that where I had added that line thinking it fixed the issue it was really just the commenting out the "PlayerName=Spotify" that fixed the issue.  So basically not actually setting Playername fixes the issue for Win10Widgets.  I have tested this with Rainmeter 4.2.0 beta r3029 and Win10Widgets v1.0.0 multiple times now.  I am using Spotify Premium if that has some bearing on this.

So to be clear for anyone reading this later:
Win10Widgets 1.0.0 (which is the current version on the site as of writing this) ships with a Spotify Plugin (also version 1.0.0) that does not currently work out of the box.  It can be fixed by navigating in the Rainmeter Manage dialog box by selecting Win10Widgets/Spotify/Spotify-Small.ini and clicking on Edit.  Then comment out the line "PlayerName=Spotify" by putting the semicolon in front of it (so that it is now "";PlayerName=Spotify").  If it doesn't work, just uncomment the original. No harm, no foul.  It is 2 seconds of effort to check, and it works for me.

Edit: I am a Spotify Premium user, so that may factor into this as well.

**Comment by BrendanTCC** [comment]
As ridiculous as it may seem, @tjhrulz, @Meowskers5 is actually correct. I tried what @Meowskers5 suggested, and it worked. Initially, the skin just would spit the "error: plugin not working" message, then it properly picked up Spotify. I'm a Spotify Free user, just for the record.

**Comment by tjhrulz** [comment]
Wow you actually stumbled upon an oddity in Rainmeter, see Rainmeter's NowPlaying plugin has a mode for an old program called Windows Live Messenger (Saying that makes me feel old) back in the day music players would forward their media status to that. So thus as a general catch all Rainmeter added support for that protocol and then sent the player commands (Which are just keyboard commands) globally (which works a lot of the time since you normally only have one media player open at a time)

See here is the thing, WLM is what Rainmeter reverts to when the player type is not recognized. Even though Spotify does not send info to WLM you use the SpotifyPlugin for all the track info. All that most people use NowPlaying for is for player bangs. Since you have only Spotify open it works fine.

Regardless I plan on working around Spotify's stupidity in the next version of Rainmeter (Not the beta that will come out likely tomorrow but the next one) now that I have finished with UsageMonitor

**Comment by BrendanTCC** [comment]
Lol... I remember when I was a younger kiddo, my little brother and I had a laptop each. His was newer - an ASUS laptop, running Windows 7 Starter. My laptop was an eMachines laptop running windows XP Home Edition. I think we both had WLM, but we couldn't get accounts because we were too young. Plus, we didn't have Wi-Fi access, because our mother only got a broadband router for her computer.

**Comment by ericabiz** [comment]
Spotify is still broken by default, though @Meowskers5 's fix works. Would be nice to see this fixed by default, though. It's been months. Any progress? Seems like a straightforward change.

**Comment by BrendanTCC** [comment]
Yeah admiredly this is starting to do my head in. *Hurry up and fix this
Spotify!* >:C
-- 
Thanks,
Brendan.

**Comment by tjhrulz** [comment]
This issue has long been worked around on our end in the beta branch of Rainmeter and is now part of the 4.2 final release, just go download that instead of using the ancient version of Rainmeter tjmarkham bundles with his skin. Also if you are using the SpotifyPlugin make sure that is up to date as well, the update for that can be found on the forums.

**Comment by BrendanTCC** [comment]
I think I've already upgraded.
-- 
Thanks,
Brendan.

**Comment by ghost** [comment]
None of the solutions on this thread are working for me.

**Comment by empz** [comment]
I've just download latest Rainmeter with Win10Widgets for the first time ever and the Spotify widget is not working.
I'm a Spotify premium user.

Can we get the fix into an official release please?

**Comment by ghost** [comment]
My fork fixes it, but you need spicetify
